File: api/src/main/java/me/devsnox/planetsystem/api/log/Logger.java
package me.devsnox.planetsystem.api.log;

public interface Logger {

    default void info(Object... object) {
        this.log(Level.INFO, object);
    }

    default void warn(Object... object) {
        this.log(Level.WARNING, object);
    }

    default void success(Object... object) {
        this.log(Level.SUCCESSFULLY, object);
    }

    void log(Level level, Object... message);

    enum Level {
        INFO,
        WARNING,
        SUCCESSFULLY
    }
}

File: api/src/main/kotlin/de/astride/planetsystem/api/log/Logger.kt
package de.astride.planetsystem.api.log

interface Logger {

    fun info(vararg any: Any) = log(Level.INFO, *any)
    fun warn(vararg any: Any) = log(Level.WARNING, *any)
    fun success(vararg any: Any) = log(Level.SUCCESSFULLY, *any)

    fun log(level: Level, vararg message: Any)

    enum class Level {
        INFO,
        WARNING,
        SUCCESSFULLY
    }

}
